### PR TITLE
Fixing Json Parsing Issue

### PR DESCRIPTION
**Pull request by vprabhakar-px**
Fixing Parsing issues with Rest APIS 

https://dogfood-skittles.dev.purestorage.com/api/login
Auth URL [https://dogfood-skittles.dev.purestorage.com/api/login]API Token [T-7a9e656d-b835-414f-88c2-0b9721992cf1]Login Success!https://dogfood-skittles.dev.purestorage.com/auth/session
Got Client details [%v] &{dogfood-skittles.dev.purestorage.com T-7a9e656d-b835-414f-88c2-0b9721992cf1   2.2  ae09d506-2b75-48f4-b100-a68c6576ad54 map[verify:true] 0xc000384060 0xc00005a5e8 0xc00005a5f0 0xc00005a5f8 0xc00005a600}
2024-05-28 23:29:16 +0530:[INFO] [flasharray.(*Client).formatPath:#270] - https://10.9.1.152/api/2.4/login
2024-05-28 23:29:16 +0530:[INFO] [flasharray.(*Client).getAuthToken:#217] - Auth URL [https://10.9.1.152/api/2.4/login]
2024-05-28 23:29:16 +0530:[INFO] [flasharray.(*Client).getAuthToken:#224] - API Token [1889260f-8858-39c9-6417-3378e9244a1b]
API Token [1889260f-8858-39c9-6417-3378e9244a1b]
2024-05-28 23:29:17 +0530:[INFO] [flasharray.(*Client).getAuthToken:#256] - Login Success!
2024-05-28 23:29:17 +0530:[INFO] [flasharray.(*Client).formatPath:#270] - https://10.9.1.152/auth/session
Got Client details [%v] &{10.9.1.152 1889260f-8858-39c9-6417-3378e9244a1b   2.4  814148bc-8186-4c94-8f30-00522b75d58d map[verify:true] 0xc0002b27b0 0xc0003980a8}
2024-05-28 23:29:18 +0530:[INFO] [flasharray.(*Client).formatPath:#270] - https://10.9.1.152/api/2.4/volumes
https://10.9.1.152/api/2.4/volumes2024-05-28 23:29:18 +0530:[INFO] [flasharray.(*Client).NewRequest:#168] - [DEBUG] key: destroyed, value: true 

2024-05-28 23:29:18 +0530:[INFO] [flasharray.(*Client).NewRequest:#174] - Base URL [https://10.9.1.152/api/2.4/volumes?destroyed=true]
2024-05-28 23:29:18 +0530:[INFO] [flasharray.(*Client).NewRequest:#191] - Adding auth token [814148bc-8186-4c94-8f30-00522b75d58d]
2024-05-28 23:29:18 +0530:[INFO] [flasharray.(*Client).Do:#124] - 
Request [&{GET https://10.9.1.152/api/2.4/volumes?destroyed=true HTTP/1.1 1 1 map[Accept:[application/json] Content-Type:[application/json] X-Auth-Token:[814148bc-8186-4c94-8f30-00522b75d58d]] {} 0x139a880 0 [] false 10.9.1.152 map[] map[] <nil> map[]   <nil> <nil> <nil> {{}}}]

&{200 OK 200 HTTP/1.1 1 1 map[Connection:[keep-alive] Content-Length:[694] Content-Type:[application/json] Date:[Tue, 28 May 2024 17:59:19 GMT] Server:[nginx] Strict-Transport-Security:[max-age=31536000; includeSubDomains;] X-Auth-Token:[814148bc-8186-4c94-8f30-00522b75d58d] X-Content-Type-Options:[nosniff] X-Frame-Options:[DENY] X-Xss-Protection:[1; mode=block]] 0xc00020ee40 694 [] false false map[] 0xc0003a0400 0xc0003a6840}[{<nil> [] false [{{0 <nil> 0 <nil> 0 0 0 0 0 0} <nil> 0 <nil> {<nil> <nil>} <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> {<nil> <nil>} {<nil> <nil>} <nil> {<nil> <nil>}}] <nil>}]

### CODE REVIEW

File: drivers/pure/flasharray/volumes.go [hunk 1]
```diff
@@ -4,28 +4,13 @@ type VolumeServices struct {
 	client *Client
 }
 
-func (fs *VolumeServices) ListAllAvailableVolumes(params map[string]string, data interface{}) ([]Volumes, error) {
-	params["destroyed"] = "false"
-	req, err := fs.client.NewRequest("GET", "volumes", params, data)
+func (vols *VolumeServices) ListAllAvailableVolumes(params map[string]string, data interface{}) ([]VolResponse, error) {
+	req, err := vols.client.NewRequest("GET", "volumes", params, data)
```
Comment (lsrinivas-pure) [inline drivers/pure/flasharray/volumes.go:8]: I see in function ListAllDestroyedVolumesFromFA you are passing params as nil. is that fine here?

File: pkg/pureutils/purefa_rest2x.go [hunk 2]
```diff
@@ -20,17 +19,21 @@ func PureCreateClientAndConnectRest226(faMgmtEndpoint string, apiToken string) (
...
 	}
 	return volumes, nil
 }
 
 // ListAllDestroyedVolumesFromFA Returns list of all Destroyed FA Volumes (Function should be used with RestAPI 2.x)
-func ListAllDestroyedVolumesFromFA(faClient *fa.Client) ([]fa.Volumes, error) {
-	volumes, err := faClient.Volumes.ListAllDestroyedVolumes(nil, nil)
+func ListAllDestroyedVolumesFromFA(faClient *flasharray.Client) ([]flasharray.VolResponse, error) {
+	params := make(map[string]string)
+	params["destroyed"] = "true"
```
Comment (lsrinivas-pure) [inline pkg/pureutils/purefa_rest2x.go:35]: is params being used some where in this function?

Comment (vprabhakar-px) [reply pkg/pureutils/purefa_rest2x.go:35]: yes, basically params are the query params that can be provided to the HTTP Request 



params map[string]string, data interface{}

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Comment by avinashsingh-px** [comment]
As discussed on slack, merged the PR.

**Pull request merged**